ambry-messageformat/src/main/java/com.github.ambry.messageformat/BlobStoreHardDelete.java

```diff
@@ -82,9 +82,13 @@ public MessageInfo getMessageInfo(Read read, long offset, StoreKeyFactory storeK
             return new MessageInfo(key, header.capacity() + key.sizeInBytes() + headerFormat.getMessageSize(),
                 Utils.addSecondsToEpochTime(properties.getCreationTimeInMs(), properties.getTimeToLiveInSeconds()));
           } else {
-            boolean deleteFlag = MessageFormatRecord.deserializeDeleteRecord(stream);
-            return new MessageInfo(key, header.capacity() + key.sizeInBytes() + headerFormat.getMessageSize(),
-                deleteFlag);
+            DeleteRecord deleteRecord = MessageFormatRecord.deserializeDeleteRecord(stream);
+            if (deleteRecord.getVersion() == MessageFormatRecord.Delete_Version_V1) {
+              return new MessageInfo(key, header.capacity() + key.sizeInBytes() + headerFormat.getMessageSize(),
+                  deleteRecord.isDeleted());
+            } else {
+              // TODO: construct new MessageInfo with accountId, containerId, operationTime
```

Comment:
I guess we could either merge the MessageInfoList PR first, or is it ok for this to throw an exception?